File: python/run_all.py
#!/usr/bin/env python3
"""
Run all Python benchmarks.
"""
import sorting
import fibonacci
import matrix
import strings
import api_requests

def main():
    print("=" * 60)
    print("Python Performance Benchmarks")
    print("=" * 60)
    print()
    
    # Sorting
    print("Running Sorting Benchmark...")
    result = sorting.run_benchmark()
    print(f"  {result['test_name']}: {result['execution_time_ms']:.2f} ms")
    print()
    
    # Fibonacci
    print("Running Fibonacci Benchmarks...")
    results = fibonacci.run_benchmark()
    for result in results:
        print(f"  {result['test_name']}: {result['execution_time_ms']:.2f} ms")
    print()
    
    # Matrix
    print("Running Matrix Multiplication Benchmark...")
    result = matrix.run_benchmark()
    print(f"  {result['test_name']}: {result['execution_time_ms']:.2f} ms")
    print()
    
    # Strings
    print("Running String Manipulation Benchmarks...")
    results = strings.run_benchmark()
    for result in results:
        print(f"  {result['test_name']}: {result['execution_time_ms']:.2f} ms")
    print()
    
    # API Requests
    print("Running API Request Benchmark...")
    result = api_requests.run_benchmark()
    print(f"  {result['test_name']}: {result['execution_time_ms']:.2f} ms")
    print(f"  Successful: {result['successful_requests']}/{result['total_requests']}")
    print()
    
    print("=" * 60)
    print("All benchmarks completed!")
    print("=" * 60)

if __name__ == '__main__':
    main()

File: ruby/run_all.rb
#!/usr/bin/env ruby

puts "=" * 60
puts "Ruby Performance Benchmarks"
puts "=" * 60
puts

# Sorting
puts "Running Sorting Benchmark..."
system("ruby sorting.rb")
puts

# Fibonacci
puts "Running Fibonacci Benchmarks..."
system("ruby fibonacci.rb")
puts

# Matrix
puts "Running Matrix Multiplication Benchmark..."
system("ruby matrix.rb")
puts

# Strings
puts "Running String Manipulation Benchmarks..."
system("ruby strings.rb")
puts

# API Requests
puts "Running API Request Benchmark..."
system("ruby api_requests.rb")
puts

puts "=" * 60
puts "All benchmarks completed!"
puts "=" * 60


